argilla-server/tests/unit/api/handlers/v1/datasets/records/records_bulk/test_create_dataset_records_bulk.py

```diff
@@ -412,3 +413,154 @@ async def test_create_dataset_records_bulk_with_image_field_data_url_exceeding_m
+
+        response = await async_client.post(
+            self.url(dataset.id),
+            headers=owner_auth_header,
+            json={
+                "items": [
+                    {
+                        "fields": {
+                            "chat": [
+                                {
+                                    "role": "user",
+                                    "content": "a",
+                                }
+                            ]
+                            * 1000,
+                        },
+                    },
+                ],
+            },
+        )
+
+        assert response.status_code == 422
+        assert "is not valid" in response.json()["detail"]
+
+        assert (await db.execute(select(func.count(Record.id)))).scalar_one() == 0
+
+    async def test_create_dataset_records_bulk_with_chat_field_with_non_dicts(
+        self, db: AsyncSession, async_client: AsyncClient, owner_auth_header: dict
+    ):
+        dataset = await DatasetFactory.create(status=DatasetStatus.ready)
+
+        await ChatFieldFactory.create(name="chat", dataset=dataset)
+        await LabelSelectionQuestionFactory.create(dataset=dataset)
+
+        response = await async_client.post(
+            self.url(dataset.id),
+            headers=owner_auth_header,
+            json={
+                "items": [
+                    {
+                        "fields": {
+                            "chat": "invalid",
+                        },
+                    },
+                ],
+            },
+        )
+
+        assert response.status_code == 422
+        assert "is not valid" in response.json()["detail"]
+
+        assert (await db.execute(select(func.count(Record.id)))).scalar_one() == 0
+
+    async def test_create_dataset_records_bulk_with_chat_field_without_role_key(
+        self, db: AsyncSession, async_client: AsyncClient, owner_auth_header: dict
+    ):
+        dataset = await DatasetFactory.create(status=DatasetStatus.ready)
+
+        await ChatFieldFactory.create(name="chat", dataset=dataset)
+        await LabelSelectionQuestionFactory.create(dataset=dataset)
+
+        response = await async_client.post(
+            self.url(dataset.id),
+            headers=owner_auth_header,
+            json={
+                "items": [
+                    {
+                        "fields": {
+                            "chat": [
+                                {
+                                    "content": "Hello!",
+                                }
+                            ],
+                        },
+                    },
+                ],
+            },
+        )
+
+        assert response.status_code == 422
+        assert "is not valid" in response.json()["detail"]
+        assert (await db.execute(select(func.count(Record.id)))).scalar_one() == 0
+
+    async def test_create_dataset_records_bulk_with_chat_field_without_content_key(
```

Comment:
I think we might need and extra check to validate they don't add extra keys. Also, I think the we might simplify the key-check tests using the parameterize over a set of keys within the json payload.

Reply:
I responded above to extra keys. Open to a PR on the feature branch for improved validation/testing.